### PR TITLE
Spring Boot 3.x integration. Fixes #485

### LINKED ISSUE #485: Spring Boot 3 integration
A Spring Boot 3 integration artifact is needed since the breaking changes between version 2 and 3 of Spring Boot.

### PR DESCRIPTION
**Pull request by rdebusscher**
Support for Spring Boot 3 which requires now the usage of AutoConfiguration and no longer spring.factories file.

Creation of a common artifact in a way

- It is compatible with current version 7.1 (no changes needed in user code, nor pom.xml)
 -No code changes required when moving from Spring Boot 2 to 3 version (just pom.xml to update)

### CODE REVIEW

**Review by fh-ms — APPROVED** [review]

**Comment by zdenek-jonas** [comment]
I would prefer to see the common as a library only, with no dependency on mIcrostream. 
Just the Configuration java classes. 
For the safety of future use, it's better if there are as few dependencies as possible.
I would leave the dependency on Microstream and other spring components to the implementation.

**Comment by rdebusscher** [comment]
@zdenek-jonas As discussed, the PR without spring-boot-common module.

File: integrations/spring-boot/pom.xml [hunk 1]
```diff
@@ -80,32 +69,6 @@
...
-                                            ${project.basedir}/target/classes/META-INF/spring-configuration-metadata.json
-                                        </file>
-                                    </files>
-                                </requireFilesExist>
-                            </rules>
-                            <fail>true</fail>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
```
Comment (zdenek-jonas) [inline integrations/spring-boot/pom.xml]: Why is this check removed?

**Comment by zdenek-jonas** [comment]
Spring must add this dependency to generate the metadata file from custom configuration fields. This file is essential for IDE (such as IntelliJ) to enable context suggestions for developers. To add
```
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-configuration-processor</artifactId>
            <version>${org.springframework.boot.version}</version>
            <scope>provided</scope>
            <optional>true</optional>
        </dependency>
```
And also please add enforce plugin to check if this file exists (like spring boot 2)

Some more information about:
https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#appendix.configuration-metadata.annotation-processor

**Review by zdenek-jonas — APPROVED** [review]

File: integrations/spring-boot/pom.xml (hunk 1 above)
Comment (rdebusscher) [reply integrations/spring-boot/pom.xml]: It seems the code is not properly updated for you. I still have this check in my version and Github says you commented on an outdated snippet.

**Pull request merged**